metabase/metabase
# feat(sdk): pivot table color customizations

Closes https://github.com/metabase/metabase/issues/42947

### Description

This PR applies the theme configuration from `theme.colors` and `theme.components` to the Pivot Table visualization in the embedding SDK, allowing for more color customizations in the SDK. The pivot table uses the table cell's colors to derive the emphasized cells and hover colors.

### How to verify

- Verify that the theming/colors in the dashboard does not change.
- Go to the `test-pivot-table` branch in the demo app.
- Open the `AppProvider.tsx` and change the settings in `theme.colors` and `theme.components`
- Go to the kitchen sink page.
- The pivot table should use the configured style accordingly

### Demo

Styling should remain the same in the dashboard.

<img src="https://github.com/metabase/metabase/assets/4714175/a96a48da-48b2-4c65-be83-0f2395387e2f" width="300">

Using "cell background" and "cell text color" to achieve white table even in apps with dark backgrounds.

<img src="https://github.com/metabase/metabase/assets/4714175/ccd876ed-d4aa-4215-a52c-9526a6df2f63" width="300">

Colored pivot table, with customized row toggle color.

<img src="https://github.com/metabase/metabase/assets/4714175/667d3b7a-6cfd-4e2f-bb53-60be7319f5ff" width="300">

When `colors.background` is set, the pivot table takes on a transparent appearance.

<img src="https://github.com/metabase/metabase/assets/4714175/07c0aa48-8fa4-4b66-a038-1501f0db5751" width="300">

### Checklist

- [x] Tests have been added/updated to cover changes in this PR (Loki - Visual Test)
